cmd/pullrequest-init/github_test.go

```diff
@@ -0,0 +1,224 @@
+	}
+	gh.AddPullRequest(pr)
+	comment := &github.IssueComment{
+		ID:   github.Int64(1),
+		Body: github.String("hello world!"),
+	}
+	gh.AddComment(owner, repo, int64(prNum), comment)
+
+	h, err := NewGitHubHandler(ctx, zap.NewNop().Sugar(), pr.GetHTMLURL())
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Override GitHub client to point to local test server.
+	h.Client = client
+
+	dir := os.TempDir()
+	if err := h.Download(ctx, dir); err != nil {
+		t.Fatal(err)
+	}
+
+	prPath := filepath.Join(dir, "pr.json")
+	rawPRPath := filepath.Join(dir, "github/pr.json")
+	rawCommentPath := filepath.Join(dir, "github/comments/1.json")
+
+	wantPR := &PullRequest{
+		Type: "github",
+		ID:   int64(prNum),
+		Head: &GitReference{
+			Repo:   pr.GetHead().GetRepo().GetCloneURL(),
+			Branch: pr.GetHead().GetRef(),
+			SHA:    pr.GetHead().GetSHA(),
+		},
+		Base: &GitReference{
+			Repo:   pr.GetBase().GetRepo().GetCloneURL(),
+			Branch: pr.GetBase().GetRef(),
+			SHA:    pr.GetBase().GetSHA(),
+		},
+		Comments: []*Comment{{
+			ID:     comment.GetID(),
+			Author: comment.GetUser().GetLogin(),
+			Text:   comment.GetBody(),
+			Raw:    rawCommentPath,
+		}},
+		Labels: []*Label{},
+		Raw:    rawPRPath,
+	}
+
+	gotPR := new(PullRequest)
+	diffFile(t, prPath, wantPR, gotPR)
+	diffFile(t, rawPRPath, pr, new(github.PullRequest))
+	if rawPRPath != gotPR.Raw {
+		t.Errorf("Raw PR path: want [%s], got [%s]", rawPRPath, gotPR.Raw)
+	}
+	diffFile(t, rawCommentPath, comment, new(github.IssueComment))
+	if rawCommentPath != gotPR.Comments[0].Raw {
+		t.Errorf("Raw PR path: want [%s], got [%s]", rawCommentPath, gotPR.Comments[0].Raw)
+	}
+
+	// Upload
+
+	// Update labels
+	gotPR.Labels = []*Label{{
+		Text: "tacocat",
+	}}
+	gotPR.Comments = append(gotPR.Comments, &Comment{
+		Text: "abc123",
+	})
+	f, err := os.Create(prPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := json.NewEncoder(f).Encode(gotPR); err != nil {
+		t.Fatal(err)
+	}
+	if err := h.Upload(ctx, dir); err != nil {
+		t.Fatal(err)
+	}
+	ghPR, _, err := client.PullRequests.Get(ctx, owner, repo, prNum)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pr.Labels = []*github.Label{{
+		Name: github.String(gotPR.Labels[0].Text),
+	}}
+	if diff := cmp.Diff(pr, ghPR); diff != "" {
+		t.Errorf("Upload PR diff: %s", diff)
+	}
+
+	ghComments, _, err := client.Issues.ListComments(ctx, owner, repo, prNum, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	wantComments := []*github.IssueComment{comment, &github.IssueComment{
+		ID:   github.Int64(2),
+		Body: github.String(gotPR.Comments[1].Text),
+	}}
+	if diff := cmp.Diff(wantComments, ghComments); diff != "" {
+		t.Errorf("Upload comment diff: %s", diff)
+	}
+
+}
+
+func diffFile(t *testing.T, path string, want interface{}, got interface{}) {
+	t.Helper()
+	f, err := os.Open(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := json.NewDecoder(f).Decode(got); err != nil {
+		t.Fatal(err)
+	}
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("PR Diff: %s", diff)
```

Comment:
nit: I think we are using `-got +wanted: %s` elsewhere

Reply:
Looks like it's `-want +got`, but done.